crates/aptos-telemetry-service/src/prometheus_push_metrics.rs

```diff
@@ -57,52 +61,82 @@ pub async fn handle_metrics_ingest(
 
     let start_timer = Instant::now();
 
-    let post_futures = client.iter().map(|(name, client)| async {
-        let result = client
-            .post_prometheus_metrics(
-                metrics_body.clone(),
-                extra_labels.clone(),
-                encoding.clone().unwrap_or_default(),
-            )
-            .await;
-
-        match result {
-            Ok(res) => {
-                METRICS_INGEST_BACKEND_REQUEST_DURATION
-                    .with_label_values(&[&claims.peer_id.to_string(), name, res.status().as_str()])
-                    .observe(start_timer.elapsed().as_secs_f64());
-                if res.status().is_success() {
-                    debug!("remote write to victoria metrics succeeded");
-                } else {
+    let mut post_futures: FuturesUnordered<_> = client
+        .into_iter()
+        .map(|(name, client)| async {
+            let result = client
```

Comment:
nit: you could use tokio::timeout, right? e.g., https://rcos.io/static/internal_docs/tokio/time/fn.timeout.html. 😄 e.g.., just use `timeout` in here, and then below, the `select!` will just wait for them all to return?

Either way, LGTM 😄

Reply:
That makes it much simpler. Thanks Josh! 

I was trying to timeout with join_all and it did not make sense, so I came up with a complex solution.